src/services/api/libs/package/getPackage.ts

```diff
@@ -1,9 +1,25 @@
 import * as os from "../../../../libs/opensearch-lib";
-import { ItemResult } from "shared-types";
+import { opensearch } from "shared-types";
 
 export const getPackage = async (id: string) => {
   if (!process.env.osDomain) {
     throw new Error("process.env.osDomain must be defined");
   }
-  return (await os.getItem(process.env.osDomain, "main", id)) as ItemResult;
+  const main = (await os.getItem(
+    process.env.osDomain,
+    "main",
+    id
+  )) as opensearch.main.ItemResult;
+  const changelog = (await os.search(process.env.osDomain, "changelog", {
+    from: 0,
+    size: 200,
+    // NOTE: get the required timestamp sort field
+    sort: [{ timestamp: "desc" }],
+    query: { bool: { must: [{ term: { "packageId.keyword": id } }] } },
+  })) as opensearch.changelog.Response;
+
+  return {
+    ...main,
+    _source: { ...main._source, changelog: changelog.hits.hits },
```

Comment:
Wondering if we should bundle them here or break them out as individual getters. With React components, we can easily load "per-component" rather than piping a top-level object thru to sub-components. 

A quick cost-benefit analysis doesn't really raise any immediate needs tho. Just starting the convo!

Reply:
I see your point.  I've pulled out the changelog getter into a separate file and moved data aggregation to the handler 👍